sklearn/utils/tests/test_class_weight.py

```diff
@@ -37,6 +38,29 @@ def test_compute_class_weight_not_present():
     assert_raises(ValueError, compute_class_weight, "auto", classes, y)
     assert_raises(ValueError, compute_class_weight, "balanced", classes, y)
 
+def test_compute_class_weight_dict():
+    classes = np.arange(3)
+    class_weights = {0: 1.0, 1: 2.0, 2: 3.0}
+    y = np.asarray([0, 0, 1, 2])
+    cw = compute_class_weight(class_weights, classes, y)
+
+    # When the user specifies class weights, compute_class_weights should just
+    # return them.
+    assert_array_almost_equal(np.asarray([1.0, 2.0, 3.0]), cw)
+
+    # When a class weight is specified that isn't in classes, a ValueError
+    # should get raised
+    msg = 'Class label 4 not present.'
+    class_weights = {0: 1.0, 1: 2.0, 2: 3.0, 4: 1.5}
+    assert_raise_message(
```

Comment:
As a nitpick, I think this is not the style that sklearn uses. I would suggest the following:

```
assert_raise_message(ValueError, msg, compute_class_weight, class_weights,
                                     classes, y)
```